src/shapes/Object/defaultValues.ts

```diff
@@ -36,75 +36,48 @@ export const cacheProperties = [
 ];
 
 export const fabricObjectDefaultValues = {
-  originX: LEFT,
-  originY: TOP,
+  // coming from ObjectOrigin
   top: 0,
   left: 0,
   width: 0,
   height: 0,
-  scaleX: 1,
-  scaleY: 1,
   flipX: false,
   flipY: false,
-  opacity: 1,
-  angle: 0,
+  scaleX: 1,
+  scaleY: 1,
   skewX: 0,
   skewY: 0,
-  cornerSize: 13,
-  touchCornerSize: 24,
-  transparentCorners: true,
-  hoverCursor: null,
-  moveCursor: null,
+  originX: LEFT,
+  originY: TOP,
+  angle: 0,
+  strokeWidth: 1,
+  strokeUniform: false,
+  // coming from ObjectGeometry
   padding: 0,
-  borderColor: 'rgb(178,204,255)',
-  borderDashArray: null,
-  cornerColor: 'rgb(178,204,255)',
-  cornerStrokeColor: '',
-  cornerStyle: 'rect',
-  cornerDashArray: null,
-  centeredScaling: false,
-  centeredRotation: true,
+  // coming from AnimatableObject
+  colorProperties: ['fill', 'stroke', 'backgroundColor'],
+  // FabricObject itself
+  minScaleLimit: 0,
+  opacity: 1,
+  paintFirst: 'fill',
   fill: 'rgb(0,0,0)',
   fillRule: 'nonzero',
-  globalCompositeOperation: 'source-over',
-  backgroundColor: '',
-  selectionBackgroundColor: '',
   stroke: null,
-  strokeWidth: 1,
   strokeDashArray: null,
   strokeDashOffset: 0,
   strokeLineCap: 'butt',
   strokeLineJoin: 'miter',
   strokeMiterLimit: 4,
+  globalCompositeOperation: 'source-over',
+  backgroundColor: '',
   shadow: null,
-  borderOpacityWhenMoving: 0.4,
-  borderScaleFactor: 1,
-  minScaleLimit: 0,
-  selectable: true,
-  evented: true,
   visible: true,
-  hasControls: true,
-  hasBorders: true,
-  perPixelTargetFind: false,
   includeDefaultValues: true,
-  lockMovementX: false,
-  lockMovementY: false,
-  lockRotation: false,
-  lockScalingX: false,
-  lockScalingY: false,
-  lockSkewingX: false,
-  lockSkewingY: false,
-  lockScalingFlip: false,
   excludeFromExport: false,
   objectCaching: true,
-  noScaleCache: true,
-  strokeUniform: false,
-  dirty: true,
-  paintFirst: 'fill',
-  activeOn: 'down',
-  colorProperties: ['fill', 'stroke', 'backgroundColor'],
   clipPath: undefined,
   inverted: false,
   absolutePositioned: false,
-  FX_DURATION: 500,
+  centeredRotation: true,
```

Comment:
interactive, related to centeredScaling

Reply:
So this one is actually used by FabricObject in the rotate method so it wouldn't make sense to move it to the subclass... unless we think the rotate method should move to InteractiveObject as well. To be honest, I'm a bit confused about the FabricObject class, because there's a FabricObject in Object.ts and a FabricObject in FabricObject.ts. The inheritance is Object > InteractiveObject > FabricObject but the class names look like FabricObject > InteractiveObject > FabricObject.

Thoughts?

Reply:
I see
You are right
I don't know what to say
Maybe we extract all transform props including centeredScaling to a standalone interface?

The rotate method should be exposed regardless of interactivity, along side scale, skew etc. but that is a different topic. I can point out where I did that.
The prototype chain of FabricObject is just a compromise we made to make an incremental TS migration from mixins to  classes.
It is not good enough and will require a major redesign of the code to change to a better composition based architecture.
So the division is actually a legacy design, a left over from the incremental TS migration. Mixins disguised as classes.

We discussed extracting stuff to managers and already started with simple ones (TextEditingManager) but the major managers have yet to be planned, e.g. caching, rendering etc.
Add to that the fact that we wanted to publish v6 a year ago and that there are 100 open prs and many more things we want to do.
I don't like this as well but since it is more internal design I guess we can live with it until we decide not to.

Hope that gave you some context.

Reply:
Thanks, it does. So you think we should have a TransformableObjectProps that ObjectProps extends and move centeredScaling to Object.ts?

Reply:
I would call it  ObjectTransformProps, yes
Then all related props reside there not needing to follow the interactie/non onteractive object legacy distinguish

Reply:
Moved centered props to ObjectTransformProps and extended base Object class to use it.